packages/edit-site/src/components/add-new-template/new-template.js

```diff
@@ -30,8 +30,40 @@ const DEFAULT_TEMPLATE_SLUGS = [
 	'index',
 ];
 
+// The "Template Hierarchy" mapping from registered templates.
+// @see https://developer.wordpress.org/themes/basics/template-hierarchy/
+const TEMPLATE_HIERARCHY = {
+	'font-page': 'home',
+	home: 'index',
+	'single-post': 'single',
+	single: 'singular',
+	singular: 'index',
+	page: 'singular',
+	archive: 'category',
+	category: 'index',
```

Comment:
I think this is slightly incorrect. A new category template should use archive as the fallback. A new archive template should use index as the fallback.

Reply:
and if neither archive or category exist, archive should fall back to index too

Reply:
BTW I just found a helper that already does something like this: https://github.com/WordPress/gutenberg/blob/2e5af3fc1f2082eabe0f113e728b18d7d3b4703f/packages/edit-site/src/components/navigation-sidebar/navigation-panel/template-hierarchy.js#L21

It's using this `const`: https://github.com/WordPress/gutenberg/blob/556248227df7c6d0fdaab6bdf9f5ff74590fb2b0/packages/edit-site/src/components/navigation-sidebar/navigation-panel/constants.js#L64-L68

We might wanna reuse that (unless we decide to go with https://github.com/WordPress/gutenberg/pull/37258 😊).